PWGJE/TableProducer/emcalCorrectionTask.cxx

```diff
@@ -237,173 +228,340 @@ struct EmcalCorrectionTask {
-        i++;
+
+        LOG(debug) << "Cluster loop done for clusterizer " << iClusterizer;
+        iClusterizer++;
       } // end of clusterizer loop
       LOG(debug) << "Done with process BC.";
       nBCsProcessed++;
-    }
+    } // end of bc loop
     LOG(detail) << "Processed " << nBCsProcessed << " BCs with " << nCellsProcessed << " cells";
   }
+  PROCESS_SWITCH(EmcalCorrectionTask, processFull, "run full analysis", true);
+
+  void processStandalone(o2::aod::BCs const& bcs, o2::aod::Collisions const& collisions, filteredCells const& cells)
+  {
+    LOG(debug) << "Starting process standalone.";
+
+    int nBCsProcessed = 0;
+    int nCellsProcessed = 0;
+    for (auto bc : bcs) {
+      LOG(debug) << "Next BC";
+      // Convert aod::Calo to o2::emcal::Cell which can be used with the clusterizer.
+      // In particular, we need to filter only EMCAL cells.
+
+      // Get the collisions matched to the BC using foundBCId of the collision
+      auto collisionsInBC = collisions.sliceBy(collisionsPerBC, bc.globalIndex());
+      auto cellsInBC = cells.sliceBy(cellsPerFoundBC, bc.globalIndex());
+
+      if (!cellsInBC.size()) {
+        LOG(debug) << "No cells found for BC";
+        countBC(bc.globalBC(), collisionsInBC.size(), false);
+        continue;
+      }
+      // Counters for BCs with matched collisions
+      countBC(bc.globalBC(), collisionsInBC.size(), true);
+      std::vector<o2::emcal::Cell> cellsBC;
+      std::vector<int64_t> cellIndicesBC;
+      for (auto& cell : cellsInBC) {
+        cellsBC.emplace_back(cell.cellNumber(),
+                             cell.amplitude(),
+                             cell.time(),
+                             o2::emcal::intToChannelType(cell.cellType()));
+        cellIndicesBC.emplace_back(cell.globalIndex());
+      }
+      LOG(detail) << "Number of cells for BC (CF): " << cellsBC.size();
+      nCellsProcessed += cellsBC.size();
+
+      fillQAHistogram(cellsBC, mHistManager);
+
+      // TODO: Helpful for now, but should be removed.
+      LOG(debug) << "Converted EMCAL cells";
+      for (auto& cell : cellsBC) {
+        LOG(debug) << cell.getTower() << ": E: " << cell.getEnergy() << ", time: " << cell.getTimeStamp() << ", type: " << cell.getType();
+      }
+
+      LOG(debug) << "Converted cells. Contains: " << cellsBC.size() << ". Originally " << cellsInBC.size() << ". About to run clusterizer.";
+
+      //  this is a test
+      //  Run the clusterizers
+      LOG(debug) << "Running clusterizers";
+      Int_t iClusterizer = 0;
+      for (auto& clusterizer : mClusterizers) {
+        clusterizer->findClusters(cellsBC);
+
+        auto emcalClusters = clusterizer->getFoundClusters();
+        auto emcalClustersInputIndices =
+          clusterizer->getFoundClustersInputIndices();
+        LOG(debug) << "Retrieved results. About to setup cluster factory.";
+
+        convertToAnalysisCluster(cellsBC, iClusterizer, emcalClusters, emcalClustersInputIndices);
```

Comment:
I would put all of this into one function cellsToClusters. Instead iterating over the clusterizers with the iterator I would use a simple for loop and pass the index to the function which internally calls the clusterizer in the vector with a given index.